/**
 * Copyright(c) 2021 Intel Corporation
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto3";

package qos;

service QosService {
    rpc Add_TM_Node(RequestAdd_TM_Node) returns (ResponseRet) {}
    rpc Set_Node_BW(RequestSet_Node_BW) returns (ResponseRet) {}
    rpc Del_Node_BW(RequestDel_Node_BW) returns (ResponseRet) {}
    rpc Get_Node_BW(RequestGet_Node_BW) returns (ResponseBW) {}
}

message RequestofPort {
    uint32 port_id = 1;
}

message RequestAdd_TM_Node {
    uint32 port_id = 1;
    uint32 profile_id = 2;
    uint32 tc_num = 3;
    uint32 vf_num = 4;
}

message RequestSet_Node_BW {
    uint32 port_id = 1;
    uint32 profile_id = 2;
    uint64 committed_bw = 3;
    uint64 peak_bw = 4;
}

message RequestDel_Node_BW {
    uint32 port_id = 1;
    uint64 profile_id = 2;
}

message RequestGet_Node_BW {
    uint32 port_id = 1;
}

message ResponseRet {
    int32 ret = 1;
    string msg = 2;
}

message ResponseBW {
    int32 ret = 1;
    uint64 committed_bw = 2;
    uint64 peak_bw = 3;
    string msg = 4;
}
